### PR TITLE
[DEPRECATED] Streaming contract state events [DPP-304]

### PR DESCRIPTION
**Pull request by kamil-da**
## Deprecated
Please see the new PR: #9365 

### Changes
- added a streaming query for contract state events
- added ingesting contract keys when indexing consuming exercise events

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [x] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/JdbcLedgerDao.scala [hunk 1]
```diff
@@ -114,6 +114,11 @@ private class JdbcLedgerDao(
   override def lookupLedgerEnd()(implicit loggingContext: LoggingContext): Future[Offset] =
     dbDispatcher.executeSql(metrics.daml.index.db.getLedgerEnd)(ParametersTable.getLedgerEnd)
 
+  // TODO
+  override def lookupLedgerEndAndEventSequentialId()(implicit
+      loggingContext: LoggingContext
+  ): Future[(Offset, Long)] = ???
+
```
Comment (kamil-da) [inline ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/JdbcLedgerDao.scala:121]: I wasn't sure on the strategy for the mutable state cache vs parallel ingestion development.
@nmarton-da @tudor-da Which package is appropriate for the mutable state cache work - `dao` or `appendonlydao`?

File: ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/TransactionsReader.scala [hunk 2]
```diff
@@ -311,6 +312,11 @@ private[appendonlydao] final class TransactionsReader(
       .watchTermination()(endSpanOnTermination(span))
   }
 
+  // TODO
+  override def getContractStateEvents(startExclusive: (Offset, Long), endInclusive: (Offset, Long))(
+      implicit loggingContext: LoggingContext
+  ): Source[((Offset, Long), ContractStateEventsReader.ContractStateEvent), NotUsed] = ???
+
```
Comment (kamil-da) [inline ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/TransactionsReader.scala:319]: The same as above.

**Review by tudor-da — CHANGES_REQUESTED** [review]

File: ledger/metrics/src/main/scala/com/daml/metrics/Metrics.scala [hunk 3]
```diff
@@ -466,6 +467,9 @@ final class Metrics(val registry: MetricRegistry) {
         val getEventSeqIdRange: DatabaseMetrics = createDbMetrics("get_event_sequential_id_range")
         val getAcsEventSeqIdRange: DatabaseMetrics =
           createDbMetrics("get_acs_event_sequential_id_range")
+        val getContractStateEvents: DatabaseMetrics = createDbMetrics(
+          "get_contract_lifecycle_events"
```
[the remark this answers is not in the document]
Comment (kamil-da) [reply ledger/metrics/src/main/scala/com/daml/metrics/Metrics.scala:471]: Fixed.

File: ledger/participant-integration-api/src/main/scala/platform/store/dao/LedgerDao.scala [hunk 4]
```diff
@@ -79,6 +79,14 @@ private[platform] trait LedgerDaoTransactionsReader {
       filter: FilterRelation,
       verbose: Boolean,
   )(implicit loggingContext: LoggingContext): Source[GetActiveContractsResponse, NotUsed]
+
+  // TODO: move this method to a separate *Reader
```
[the remark this answers is not in the document]
Comment (kamil-da) [reply ledger/participant-integration-api/src/main/scala/platform/store/dao/LedgerDao.scala:83]: You're right. I've deleted the comment.

File: ledger/participant-integration-api/src/main/scala/platform/store/dao/MeteredLedgerDao.scala [hunk 5]
```diff
@@ -48,6 +48,10 @@ private[platform] class MeteredLedgerReadDao(ledgerDao: LedgerReadDao, metrics:
   override def lookupLedgerEnd()(implicit loggingContext: LoggingContext): Future[Offset] =
     Timed.future(metrics.daml.index.db.lookupLedgerEnd, ledgerDao.lookupLedgerEnd())
 
+  override def lookupLedgerEndAndEventSequentialId()(implicit
+      loggingContext: LoggingContext
+  ): Future[(Offset, Long)] = ledgerDao.lookupLedgerEndAndEventSequentialId()
```
[the remark this answers is not in the document]
Comment (kamil-da) [reply ledger/participant-integration-api/src/main/scala/platform/store/dao/MeteredLedgerDao.scala:53]: Done.

File: ledger/participant-integration-api/src/main/scala/platform/store/dao/ParametersTable.scala [hunk 6]
```diff
@@ -46,6 +59,16 @@ private[dao] object ParametersTable {
 
   private val SelectLedgerEnd: SimpleSql[Row] = SQL"select #$LedgerEndColumnName from #$TableName"
 
+  // TODO: IMPLEMENT A PROPER SOLUTION
+  // THIS IS A SIMPLE WORKAROUND FOR THE LACK OF THE SEQUENTIAL ID IN THE `parameters` TABLE.
+  // THE SEQUENTIAL ID IS GOING TO BE INTRODUCED AS A PART OF THE APPEND-ONLY SCHEMA
```
[the remark this answers is not in the document]
Comment (kamil-da) [reply ledger/participant-integration-api/src/main/scala/platform/store/dao/ParametersTable.scala:64]: Currently my changes are integrated in the base `dao` package. Do I understand correctly that there is no need to integrate them with `appendonlydao` package for now?

**Pull request closed without merging**